### ISSUE TITLE
TypeError: Object(...) is not a function

### ISSUE THREAD

**Issue by ratsey** [kmathy/cerializr#1]
Quite probably a missed step on my part, but can't get past this error.

```
import { autoserializeAs, serializeAs, deserializeAs } from 'cerializr';
export class Starship {
    @serializeAs(String)
    @deserializeAs(Number)
    remainingFuel : number;
}
```

```
  public testJSON() {
    let test = {
      remainingFuel: 500.5,
    }
    return Deserialize(test, Starship);
  }
```

`"experimentalDecorators": true,`

Gives me a `TypeError: Object(...) is not a function` error against the @serializeAs(String) line in Starship. Is there anything specific to get it to recognize the decorators??

**Comment by kmathy** [comment]
Hi,

Thank you for reporting this issue.

It seems there is a problem with the webpack config. I'm trying to enable decorators with webpack + babel, but I have no significant result for now...

Please use cerializr@3.0.3, which is not using webpack.

I will update you when I fixed it ;)

**Comment by ratsey** [comment]
That worked - thanks. Excited that you're picking up on this library. I feel that it's a vitally missing core function.

**Comment by kmathy** [comment]
Should be fixed now with 3.1.2.

The problem is Webpack + Babel having problems to handle decorators. There are plugins available, but not for class properties with arguments.

Maybe when the decorators will be stable, there will be a solution, but for now, I have to work differently to generate the bundle.

I will take a look to optimize the output. Should be able to use Gulp or Grunt (back to oldies!)
